<!doctype html>
<html class="no-js" lang="en-US">
  <head>
    <meta charset="utf-8">
    <title>Bioinformatics with R | Muhammad Yaseen</title>
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link href="https://fonts.googleapis.com/css?family=Fira+Sans|Merriweather|Fira+Mono" rel="stylesheet">
    <link rel="stylesheet" href="/css/foundation.min.css">
    <link rel="stylesheet" href="/css/highlight.min.css">
    <link rel="stylesheet" href="/css/font-awesome.min.css">
    <link rel="stylesheet" href="/css/academicons.min.css">
    <link rel="stylesheet" href="/css/finite.css">
    <link rel="stylesheet" href="/css/kube.css">
    <link rel="stylesheet" href="/css/myaseen208.css">
  </head>
  <body>
    <header>
      <nav class="nav-bar">
	
	  <div class="title-bar" data-responsive-toggle="site-menu" data-hide-for="medium">
	    <button class="site-hamburger" type="button" data-toggle>
	      <i class="fa fa-bars fa-lg" aria-hidden="true"></i>
	    </button>
	    <div class="title-bar-title site-title">
	      <a href="/"  style="color: #b3cde0">Muhammad Yaseen</a>
	    </div>
	    <div class="title-bar-right pull-right">
	      


	    </div>
	  </div>

	  
	    <div class="top-bar" id="site-menu" >
	      <div class="top-bar-title show-for-medium site-title">
		<a href="/"  style="color: #b3cde0;">Muhammad Yaseen</a>
	      </div>
	      <div class="top-bar-left">
		<ul class="menu vertical medium-horizontal">
		  
		  
		  <li><a href="/myaseen208/">myaseen208 blog</a></li>
		  
		  <li><a href="/publications/">Publications</a></li>
		  
		  <li><a href="/software/">Software</a></li>
		  
		  <li><a href="/seminars/">Seminars</a></li>
		  
		  <li><a href="/teaching/">Teaching</a></li>
		  
		  <li><a href="/about/">About</a></li>
		  
		</ul>
	      </div>
	      <div class="top-bar-right show-for-medium">
		


	      </div>
	    </div>
	  
	</nav>
    </header>
    <main>
      

<div class="wrapper">
<div class="units-row" style='margin-bottom: 0px;'>
<div class="unit-20">&nbsp;</div>
<div class="unit-60"><h1>Bioinformatics with R</h1></div>
</div>
<div class="units-row">
  <div class="unit-20 dateblock" style='text-align: right;'><h4 style='padding: .8ex 0px .8ex 0px;'><a href="/myaseen208">myaseen208</a></h4>
      <div class="post-metadata">
  <span class="post-date">
    <time datetime="2018-09-20 00:00:00 &#43;0000 UTC" itemprop="datePublished">20 September 2018</time>
  </span>
  
  
  
  <span class="post-tags">
    <p><i class="fa fa-tags"></i>
    
    <a class="post-tag" href="/categories/r">R</a>,
     
    
    
    <a class="post-tag" href="/categories/statistics">Statistics</a>, 
    
    
    <a class="post-tag" href="/categories/bioinformatics">Bioinformatics</a>
    
  </span>
  
  
</div>

    </div>
<div class="unit-60">

  <div id="discrete-data-presentation" class="section level1">
<h1>Discrete Data Presentation</h1>
<p>Following data presents the number of nucleotides of gene sequence (A, C, G, T). This is illustrated by the Zyxin gene which plays an important role in cell adhesion (Golub et al, 1999). The accession number (X94991.1) of one of its variants can be found in a data base like NCBI (UniGene). Given data will be used to illustrate the construction of pie chart from the frequency table of four nucleotides.</p>
<table>
<thead>
<tr class="header">
<th>A</th>
<th>C</th>
<th>G</th>
<th>T</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>410</td>
<td>789</td>
<td>573</td>
<td>394</td>
</tr>
</tbody>
</table>
<p>Data from the GenBank can also be imported directly by the following code.</p>
<pre class="r"><code># install.packages(pkgs = &quot;ape&quot;, repo = &quot;http://cran.r-project.org&quot;, dependencies =TRUE)
# library(ape)
# df1 &lt;- table(read.GenBank(c(&quot;X94991.1&quot;),as.character=TRUE))
# df1</code></pre>
<div id="simple-bar-chart" class="section level2">
<h2>Simple Bar Chart</h2>
<pre class="r"><code>if (!require(&quot;tidyverse&quot;)) install.packages(&quot;tidyverse&quot;)
# library(tidyverse)
df2 &lt;- tibble(
    G.Sequence   = c(&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;)
  , Nucleotides  = c(410, 789, 573, 394)
  )
df2</code></pre>
<pre><code># A tibble: 4 x 2
  G.Sequence Nucleotides
  &lt;chr&gt;            &lt;dbl&gt;
1 A                  410
2 C                  789
3 G                  573
4 T                  394</code></pre>
<pre class="r"><code>ggplot(
        data = df2
      , mapping = aes(x = G.Sequence, y = Nucleotides)) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-1-1.svg" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="two-way-contingency-table" class="section level1">
<h1>Two Way Contingency Table</h1>
<p>Source: OMB Statistical Policy Working Paper 22. <a href="https://www.hhs.gov/sites/default/files/spwp22.pdf" class="uri">https://www.hhs.gov/sites/default/files/spwp22.pdf</a>
Following data set consists of information concerning delinquent children. Recorded variables are Number of Delinquent Children by County and Education Level of Household Head.</p>
<pre class="r"><code>df3 &lt;- tibble(
  Delinquent = gl(n = 4, k = 4, length = 16, labels = c(&quot;Alpha&quot;,&quot;Beta&quot;,&quot;Gamma&quot;,&quot;Delta&quot;))
, EduLevel   = gl(n = 4, k = 1, length = 16, labels = c(&quot;Low&quot;,&quot;Medium&quot;,&quot;High&quot;,&quot;Very High&quot;))
, Freq       = c(15, 0, 5, 0, 20, 10, 10, 15, 5, 10, 10, 0, 10, 15, 5, 5)
)

df3</code></pre>
<pre><code># A tibble: 16 x 3
   Delinquent EduLevel   Freq
   &lt;fct&gt;      &lt;fct&gt;     &lt;dbl&gt;
 1 Alpha      Low          15
 2 Alpha      Medium        0
 3 Alpha      High          5
 4 Alpha      Very High     0
 5 Beta       Low          20
 6 Beta       Medium       10
 7 Beta       High         10
 8 Beta       Very High    15
 9 Gamma      Low           5
10 Gamma      Medium       10
11 Gamma      High         10
12 Gamma      Very High     0
13 Delta      Low          10
14 Delta      Medium       15
15 Delta      High          5
16 Delta      Very High     5</code></pre>
<pre class="r"><code>ggplot(
       data = df3
    ,  mapping = aes(x = Delinquent, y = Freq)) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-3-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(
    data = df3
  , mapping = aes(x = EduLevel, y = Freq)) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-4-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(
       data = df3
     , mapping = aes(x = Delinquent, y = Freq, fill = EduLevel)) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-5-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(
        data = df3
      , mapping = aes(x = Delinquent, y = Freq, fill = EduLevel)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-6-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(
        data = df3
      , mapping = aes(x = EduLevel, y = Freq, fill = Delinquent)) +
  geom_bar(stat = &quot;identity&quot;,position = &quot;dodge&quot;) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-7-1.svg" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="continuous-data-presentation" class="section level1">
<h1>Continuous Data Presentation</h1>
<p>The golub table contains gene expression values from 3051 genes taken from 38 Leukemia patients. Twenty seven patients are diagnosed as acute lymphoblastic leukemia (ALL) and eleven as acute myeloid leukemia (AML). The golub.gnames table contains information on the gene, including gene index, manufacturing ID, and biological name. Following table presents the gene expression value by their tumor type.</p>
<table>
<thead>
<tr class="header">
<th>tumortype</th>
<th>genevalue</th>
<th>tumortype</th>
<th>genevalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ALL</td>
<td>2.10892</td>
<td>ALL</td>
<td>1.78352</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.52405</td>
<td>ALL</td>
<td>0.45827</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>1.96403</td>
<td>ALL</td>
<td>2.18119</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>2.33597</td>
<td>ALL</td>
<td>2.31428</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>1.85111</td>
<td>ALL</td>
<td>1.99927</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.99391</td>
<td>ALL</td>
<td>1.36844</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>2.06597</td>
<td>ALL</td>
<td>2.37351</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.81649</td>
<td>ALL</td>
<td>1.83485</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>2.17622</td>
<td>AML</td>
<td>0.88941</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.80861</td>
<td>AML</td>
<td>1.45014</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>2.44562</td>
<td>AML</td>
<td>0.42904</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.90496</td>
<td>AML</td>
<td>0.82667</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>2.76610</td>
<td>AML</td>
<td>0.63637</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.32551</td>
<td>AML</td>
<td>1.02250</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>2.59385</td>
<td>AML</td>
<td>0.12758</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.92776</td>
<td>AML</td>
<td>-0.74333</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>1.10546</td>
<td>AML</td>
<td>0.73784</td>
</tr>
<tr class="even">
<td>ALL</td>
<td>1.27645</td>
<td>AML</td>
<td>0.49470</td>
</tr>
<tr class="odd">
<td>ALL</td>
<td>1.83051</td>
<td>AML</td>
<td>1.12058</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df4 &lt;- tibble(
   genevalue = c(
              2.10892, 1.52405, 1.96403, 2.33597, 1.85111, 1.99391
            , 2.06597, 1.81649, 2.17622, 1.80861, 2.44562, 1.90496
            , 2.76610, 1.32551, 2.59385, 1.92776, 1.10546, 1.27645
            , 1.83051, 1.78352, 0.45827, 2.18119, 2.31428, 1.99927
            , 1.36844, 2.37351, 1.83485, 0.88941, 1.45014, 0.42904
            , 0.82667, 0.63637, 1.02250, 0.12758, -0.74333, 0.73784
            , 0.49470, 1.12058
            )
  , tumortype = rep(c(&quot;ALL&quot;,&quot;AML&quot;), c(27, 11))
)

df4</code></pre>
<pre><code># A tibble: 38 x 2
   genevalue tumortype
       &lt;dbl&gt; &lt;chr&gt;    
 1      2.11 ALL      
 2      1.52 ALL      
 3      1.96 ALL      
 4      2.34 ALL      
 5      1.85 ALL      
 6      1.99 ALL      
 7      2.07 ALL      
 8      1.82 ALL      
 9      2.18 ALL      
10      1.81 ALL      
# ... with 28 more rows</code></pre>
</div>
<div id="frequency-distribution" class="section level1">
<h1>Frequency Distribution</h1>
<pre class="r"><code>df5 &lt;- df4 %&gt;% 
  summarize(
    R = max(genevalue) - min(genevalue)
  , k = floor(1 + 3.3*log10(length(genevalue)))
  , h = R/k
    )

df4Freq &lt;- df4 %&gt;% 
  mutate(
          Classes = cut(
                         x              = genevalue
                       , breaks         = df5$k
                       , include.lowest = TRUE
                       , right          = FALSE
                       )
          ) %&gt;%
  count(Classes) %&gt;% 
  tidyr::separate(col = Classes, into = c(&quot;LB&quot;, &quot;UB&quot;), sep = &quot;,&quot;, remove = FALSE) %&gt;%
  rename(f = n) %&gt;%
  mutate(
    LB = readr::parse_number(x = LB)
  , UB = readr::parse_number(x = UB)
  , rf = f/sum(f)
  , pf = f/sum(f)*100
  , cf = cumsum(f)
  , MidPoint = (LB + UB)/2
  ) 
  
df4Freq</code></pre>
<pre><code># A tibble: 6 x 8
  Classes             LB     UB     f     rf    pf    cf MidPoint
  &lt;fct&gt;            &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt;
1 [-0.747,-0.158) -0.747 -0.158     1 0.0263  2.63     1   -0.452
2 [-0.158,0.426)  -0.158  0.426     1 0.0263  2.63     2    0.134
3 [0.426,1.01)     0.426  1.01      7 0.184  18.4      9    0.718
4 [1.01,1.6)       1.01   1.6       8 0.211  21.1     17    1.31 
5 [1.6,2.18)       1.6    2.18     15 0.395  39.5     32    1.89 
6 [2.18,2.77]      2.18   2.77      6 0.158  15.8     38    2.48 </code></pre>
<div id="histogram" class="section level2">
<h2>Histogram</h2>
<pre class="r"><code>ggplot(
       data = df4
     , mapping = aes(x = genevalue)) + 
  geom_histogram(color = &quot;blue&quot;, fill = &quot;white&quot;, binwidth = 0.25) +
  theme_bw()+
  scale_y_continuous(expand = c(0, 0)) +
  labs(title = &quot;Histogram for Gene Value&quot;, x = &quot;Gene Value&quot;, y = &quot;Count&quot;)</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-10-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = df4Freq, mapping = aes(x = MidPoint, y = f)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.5, colour = &quot;black&quot;, fill = &quot;white&quot;) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-11-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = df4Freq, mapping = aes(x = MidPoint, y = f))  + 
  geom_point() + 
  geom_line() + 
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-12-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = df4Freq, mapping = aes(x = MidPoint, y = rf)) + 
  geom_point() + 
  geom_line() + 
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-13-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = df4Freq, mapping = aes(x = MidPoint, y = pf)) + 
  geom_point() + 
  geom_line() +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-14-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = df4Freq, mapping = aes(x = MidPoint, y = cf)) + 
  geom_point()+ 
  geom_line() +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-15-1.svg" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="stem-and-leaf-plot" class="section level2">
<h2>Stem and Leaf Plot</h2>
<pre class="r"><code>stem(x = df4$genevalue, scale = 1, width = 80, atom = 1e-08)</code></pre>
<pre><code>
  The decimal point is at the |

  -0 | 7
  -0 | 
   0 | 14
   0 | 556789
   1 | 011334
   1 | 5588888999
   2 | 00011223344
   2 | 68</code></pre>
</div>
<div id="box-plot" class="section level2">
<h2>Box Plot</h2>
<pre class="r"><code>ggplot(data = df4 , aes(x = tumortype, y = genevalue)) + 
  geom_boxplot()+
  theme_bw()</code></pre>
<p><img src="/myaseen208/2018-09-20_BioInfoR_files/figure-html/unnamed-chunk-17-1.svg" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="measures-of-centeral-tendency" class="section level1">
<h1>Measures of Centeral Tendency</h1>
<pre class="r"><code>df4 %&gt;%
  group_by(tumortype) %&gt;%
  summarize(
            n       = length(genevalue)
          , Mean    = mean(genevalue)
          , Median  = median(genevalue)
          , Minimum = min(genevalue) 
          , Maximum = max(genevalue) 
          , Q1      = quantile(x = genevalue, probs = 0.25)
          , Q2      = quantile(x = genevalue, probs = 0.50)
          , Q3      = quantile(x = genevalue, probs = 0.75)
  )</code></pre>
<pre><code># A tibble: 2 x 9
  tumortype     n  Mean Median Minimum Maximum    Q1    Q2    Q3
  &lt;chr&gt;     &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
1 ALL          27 1.89   1.93    0.458    2.77 1.80  1.93  2.18 
2 AML          11 0.636  0.738  -0.743    1.45 0.462 0.738 0.956</code></pre>
</div>
<div id="measures-of-dispersion" class="section level1">
<h1>Measures of Dispersion</h1>
<pre class="r"><code>df4 %&gt;%
  group_by(tumortype) %&gt;%
  summarize(
            IQR      = IQR(genevalue)
          , Variance = var(genevalue)
          ,  SD      = sd(genevalue)
          )</code></pre>
<pre><code># A tibble: 2 x 4
  tumortype   IQR Variance    SD
  &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;
1 ALL       0.383    0.241 0.491
2 AML       0.494    0.338 0.582</code></pre>
</div>
<div id="measures-of-skewness" class="section level1">
<h1>Measures of Skewness</h1>
<pre class="r"><code>df4 %&gt;% 
  group_by(tumortype) %&gt;%
  summarize(
           SK = sum((genevalue - mean(genevalue))^3)/(n()*(sd(genevalue))^3)
          )</code></pre>
<pre><code># A tibble: 2 x 2
  tumortype     SK
  &lt;chr&gt;      &lt;dbl&gt;
1 ALL       -0.802
2 AML       -0.937</code></pre>
</div>
<div id="measures-of-skewness-1" class="section level1">
<h1>Measures of Skewness</h1>
<pre class="r"><code>df4 %&gt;% 
  group_by(tumortype) %&gt;%
  summarize(
            K = sum((genevalue - mean(genevalue))^4)/(n()*(sd(genevalue))^4) - 3
          )</code></pre>
<pre><code># A tibble: 2 x 2
  tumortype     K
  &lt;chr&gt;     &lt;dbl&gt;
1 ALL       0.855
2 AML       0.345</code></pre>
</div>


      <meta itemprop="wordCount" content="1338">
      <meta itemprop="datePublished" content="2018-09-20">
      <meta itemprop="url" content="/myaseen208/2018-bioinfor/">

<div class='sans' style="background: #dddddd; padding: 2em;">
<div style="padding-bottom: 1em" >
<div class="share-box">
  <ul class="share">
    <li>
      <a class="facebook" href="http://www.facebook.com/sharer.php?u=%2fmyaseen208%2f2018-bioinfor%2f" target="_blank">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="twitter" href="https://twitter.com/intent/tweet?text=Bioinformatics%20with%20R. %2fmyaseen208%2f2018-bioinfor%2f from @myaseen208" target="_blank">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="linkedin" href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2fmyaseen208%2f2018-bioinfor%2f&amp;title=Bioinformatics%20with%20R" target="_blank">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo" href="http://service.weibo.com/share/share.php?url=%2fmyaseen208%2f2018-bioinfor%2f&amp;title=Bioinformatics%20with%20R" target="_blank">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email" href="mailto:?subject=Bioinformatics%20with%20R&amp;body=%2fmyaseen208%2f2018-bioinfor%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


</div>

    <ul class="pagination" role="navigation" aria-label="Pagination">
      
      <li class="arrow" aria-disabled="true"><a href="/myaseen208/2018-08-16-pastic-ntu/">&laquo; <em>Previous<span class="show-for-sr"> page</span></em>: An Introduction to Statistics using SPSS</a></li>
      
      
    </ul>
    <a href="/myaseen208/">All myaseen208 posts by date</a>
</div>

<span style='font-family: Carlito'>
 <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    if (window.location.hostname == "localhost")
              return;

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = '<https-github-com-myaseen208-myaseen208>';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>

<div id="disqus_thread"></div>
<script>






(function() { 
var d = document, s = d.createElement('script');
s.src = 'https://myaseen208.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
                            

</span>

  </div>
  </div></div>

    </main>
    <footer class="whatisthis">
  <hr class='separator'/>
  <span style="font-family: Carlito, sans; font-size: 0.9em; line-height: 1.05em;">
  <div class="row">
    <div class="column small-12 medium-4">
      <figure class='image-left'>
      <a href="/" style="border-bottom: none;">
      	<img src="/img/myaseen208.jpg" alt="Portrait of the author" class='avatar' style="max-width: 50%;">
      </a>
  </figure>
  <p>
Muhammad Yaseen is Assistant Professor of Statistics at <a href=http://www.uaf.edu.pk/>University of Agriculture Faisalabad</a> , Faisalabad 38000, Pakistan.  </p>
    </div>
    <div class="column small-12 medium-4">
      
      <h4 id="contact">Contact</h4>
      <ul class="fa-ul">
	
	<li><i class="fa-li fa fa-university" style="color:#03396c;font-size:80%;padding-top:6px;"></i><a href="http://www.uaf.edu.pk/EmployeeDetail.aspx?userid=281">Department of Mathematics &amp; Statistics, University of Agriculture Faisalabad, Faisalabad 38000, Pakistan.</a></li>
	
	<li><i class="fa-li fa fa-envelope" style="color:#03396c;font-size:80%;padding-top:6px;"></i><a href="mailto:myaseen208@gmail.com">myaseen208@gmail.com</a></li>
	
	<li><i class="fa-li fa fa-twitter" style="color:#03396c;font-size:80%;padding-top:6px;"></i><a href="https://twitter.com/myaseen208">@myaseen208</a> on Twitter</li>
	
	<li><i class="fa-li fa fa-github-alt" style="color:#03396c;font-size:80%;padding-top:6px;"></i><a href="https://github.com/myaseen208/">myaseen208</a> on GitHub</li>
	
	<li><i class="fa-li ai ai-google-scholar" style="color:#03396c;font-size:80%;padding-top:6px;"></i><a href="https://scholar.google.com.pk/citations?user=mHkR-7cAAAAJ&amp;hl=en">Google Scholar</a></li>
	
      </ul>
    </div>
    <div class="column small-12 medium-4">
      <h4>Subscribe</h4>
      <p><i><i class="fa fa-rss-square" style="color:#03396c;"></i> To receive updates from this site, you can subscribe to using an <a href="http://feeds.feedburner.com/myaseen208">RSS feed reader</a> or <a href="https://feedburner.google.com/fb/a/mailverify?uri=myaseen208&amp;loc=en_US">by email</a>.</i></p>
      <h4>Search</h4>
      <a id="searchsite">
	<form method="get" action="https://duckduckgo.com/">
	  <label for="search-field" class="show-for-sr">Search the site</label>
	  <input type="search" name="q" maxlength="255" placeholder="Search the site" id="search-field">
	  <input type="hidden" name="sites" value="https://myaseen208.com"/>
	  <input type="hidden" name="k7" value="#faf8f8"/>
	  <input type="hidden" name="kj" value="#b33"/>
	  <input type="hidden" name="ky" value="#fafafa"/>
	  <input type="hidden" name="kx" value="b"/>
	  <input type="hidden" name="ko" value="-1"/>
	  <input type="hidden" name="k1" value="-1"/>
	  <input type="submit" value="DuckDuckGo Search" style="visibility: hidden;" />
	</form>
      </a>
    </div>
  </div>
  </span>
</footer>

    <div class="endofpage">
    <footer id="footer">
    <section class="wrapper small">Powered by <a href="http://gohugo.io">Hugo</a>
    and <a href="https://github.com/rstudio/blogdown">Blogdown</a>.
     &copy; <a href="https://myaseen208.com">Muhammad Yaseen</a> 2012&ndash;2018.
    </section>
    </footer>
    </div>

    <script src="/js/jquery.js"></script>
    <script src="/js/what-input.js"></script>
    <script src="/js/foundation.min.js"></script>
    <script src="/js/finite.js"></script>

    
    <script src="/js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>

    
    
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
      tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}
      });
    </script>
    <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS_CHTML">
    </script>
    


<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-5004894-1', 'auto');
ga('send', 'pageview');
</script>

 </body>
</html>
